Common/Data/Market/VolumeRenkoBar.cs

```diff
@@ -0,0 +1,143 @@
+        /// </summary>
+        public decimal Volume { get; private set; }
+
+        /// <summary>
+        /// Gets the end time of this renko bar or the most recent update time if it <see cref="IsClosed"/>
+        /// </summary>
+        public override DateTime EndTime { get; set; }
+        
+        /// <summary>
+        /// Gets the time this bar started
+        /// </summary>
+        public DateTime Start
+        {
+            get { return Time; }
+            private set { Time = value; }
+        }
+
+        /// <summary>
+        /// Gets whether or not this bar is considered closed.
+        /// </summary>
+        public bool IsClosed => Volume >= _brickSize;
+
+        /// <summary>
+        /// Gets the kind of the bar
+        /// </summary>
+        public RenkoType Type { get; private set; }
+
+        /// <summary>
+        /// Updates this <see cref="VolumeRenkoBar"/> with the specified values and returns whether or not this bar is closed
+        /// </summary>
+        /// <param name="symbol">symbol of the data</param>
+        /// <param name="start">The current data start time</param>
+        /// <param name="endTime">The current data end time</param>
+        /// <param name="brickSize">The preset volume capacity of this bar</param>
+        /// <param name="open">The current data open value</param>
+        /// <param name="high">The current data high value</param>
+        /// <param name="low">The current data low value</param>
+        /// <param name="close">The current data close value</param>
+        /// <param name="volume">The current data volume</param>
+        public VolumeRenkoBar(Symbol symbol, DateTime start, DateTime endTime, decimal brickSize, decimal open, decimal high, decimal low, decimal close, decimal volume)
+        {
+            Type = RenkoType.Classic;
+            _brickSize = brickSize;
+
+            Symbol = symbol;
+            Start = start;
+            EndTime = endTime;
+            Open = open;
+            Close = close;
+            Volume = volume;
+            High = high;
+            Low = low;
+        }
+
+        /// <summary>
+        /// Updates this <see cref="VolumeRenkoBar"/> with the specified values and returns whether or not this bar is closed
+        /// </summary>
+        /// <param name="time">The current data end time</param>
+        /// <param name="high">The current data high value</param>
+        /// <param name="low">The current data low value</param>
+        /// <param name="close">The current data close value</param>
+        /// <param name="volume">The current data volume</param>
+        /// <returns>The excess volume that the current bar cannot absorb</returns>
+        public decimal Update(DateTime time, decimal high, decimal low, decimal close, decimal volume)
+        {
+            if (Type == RenkoType.Wicked)
+                throw new InvalidOperationException("A \"Wicked\" RenkoBar cannot be updated!");
```

Comment:
This would never happen by definition? This isn't an argument `Type = RenkoType.Classic;` it shouldn't even be part of the class? does a Wicked Volumen Renko Bar exist?